plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py

```diff
@@ -231,171 +286,121 @@ def read_next_set_of_high_ber_or_pdr_ports(self, endpoint_port):
+            port_obj.last_timestamp = timestamp
             if cable_temp is not None and not numpy.isnan(cable_temp):
                 if cable_temp == "NA" or cable_temp == "N/A" or cable_temp == "" or cable_temp == "0C":
                     continue
                 cable_temp = int(cable_temp.split("C")[0]) if type(cable_temp) == str else cable_temp
-                dT = abs(self.ports_data[port_name].get(Constants.TEMP_COUNTER, 0) - cable_temp)
-                self.ports_data[port_name][Constants.TEMP_COUNTER] = cable_temp
+                dT = abs(port_obj.counters_values.get(Constants.TEMP_COUNTER, 0) - cable_temp)
+                port_obj.counters_values[Constants.TEMP_COUNTER] = cable_temp
             if rcv_pkt_rate and error_rate / rcv_pkt_rate > self.max_pdr:
                 issues[port_name] = Issue(port_name, Constants.ISSUE_PDR)
-            elif cable_temp and (cable_temp > self.tmax or dT > self.d_tmax):
+            elif self.temp_check and cable_temp and (cable_temp > self.tmax or dT > self.d_tmax):
                 issues[port_name] = Issue(port_name, Constants.ISSUE_OONOC)
+            elif self.no_down_count and link_downed_rate > 0:
+                link_downed_issue = self.check_link_down_condition(port_obj, port_name, ports_counters)
+                if link_downed_issue:
+                    issues[port_name] = link_downed_issue
             if self.configured_ber_check:
-                error_lane_0 = get_counter(Constants.PHY_RAW_ERROR_LANE0, row, default=None)
-                error_lane_1 = get_counter(Constants.PHY_RAW_ERROR_LANE1, row, default=None)
-                error_lane_2 = get_counter(Constants.PHY_RAW_ERROR_LANE2, row, default=None)
-                error_lane_3 = get_counter(Constants.PHY_RAW_ERROR_LANE3, row, default=None)
-                if error_lane_0 and error_lane_1 and error_lane_2 and error_lane_3:
-                    symbol_ber_val = sum([
-                        get_counter(Constants.PHY_RAW_ERROR_LANE0, row),
-                        get_counter(Constants.PHY_RAW_ERROR_LANE1, row),
-                        get_counter(Constants.PHY_RAW_ERROR_LANE2, row),
-                        get_counter(Constants.PHY_RAW_ERROR_LANE3, row),
-                        ])
-                else:
-                    symbol_ber_val = None
-                eff_ber_val = get_counter(Constants.PHY_EFF_ERROR, row, default=None)
-                raw_ber_val = get_counter(Constants.RAW_BER, row, default=None)
-                #DEBUG
-                # if self.iteration < 2:
-                #     symbol_ber_val = 10
-                # else:
-                #     symbol_ber_val = 40000
-                #DEBUG
-                if symbol_ber_val is not None or eff_ber_val is not None or raw_ber_val is not None:
-                    curr_timestamp = int(time.time())
+                symbol_ber_val = calc_symbol_ber_val(row)
+                if symbol_ber_val is not None:
```

Comment:
is not None?

 if symbol_ber_val: will not work?

Reply:
no because it will return False if symbol_ber_val equals 0 and i don't want to skip if the value is 0